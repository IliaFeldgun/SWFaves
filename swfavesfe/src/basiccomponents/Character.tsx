import React from 'react';
import { ICharacter } from "../interfaces/ICharacter"

export class Character extends React.PureComponent<{character: ICharacter}, {}>{
    render(){
        return (
            <h2 className="character-name">
                {this.props.character.name}
            </h2>
        )
    }
}