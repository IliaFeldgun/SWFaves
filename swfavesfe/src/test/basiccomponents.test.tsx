import React from 'react';
import {Character} from '../basiccomponents/Character'
import renderer from 'react-test-renderer'

it('renders a character', () => {
  const character = renderer.create(<Character character={{name: "Ilusha"}} />).toJSON();

  expect(character).toMatchSnapshot();
});