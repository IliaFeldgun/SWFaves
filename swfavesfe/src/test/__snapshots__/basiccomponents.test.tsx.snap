// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a character 1`] = `
<h2
  className="character-name"
>
  Ilusha
</h2>
`;
